File: backend/internal/server/routes/sitenav.go
package routes

import (
	"micro-net-hub/internal/server/handler"
	"micro-net-hub/internal/server/middleware"

	jwt "github.com/appleboy/gin-jwt/v2"
	"github.com/gin-gonic/gin"
)

func InitSiteNavRoutes(r *gin.RouterGroup, authMiddleware *jwt.GinJWTMiddleware) gin.IRoutes {
	var h handler.SiteNavHandler

	sitenav := r.Group("/sitenav")
	{
		sitenav.GET("/getnav", h.GetNav)
	}

	group := sitenav.Group("/group")
	group.Use(authMiddleware.MiddlewareFunc())
	group.Use(middleware.CasbinMiddleware())
	{
		group.POST("/add", h.AddNavGroup)
		group.POST("/update", h.UpdateNavGroup)
		group.POST("/delete", h.DeleteNavGroup)
	}

	site := sitenav.Group("/site")
	site.Use(authMiddleware.MiddlewareFunc())
	site.Use(middleware.CasbinMiddleware())
	{
		site.POST("/add", h.AddNavSite)
		site.POST("/update", h.UpdateNavSite)
		site.POST("/delete", h.DeleteNavSite)
	}
	return r
}

File: frontend/src/api/sitenav/sitenav.js
import request from "@/utils/request";
// 获取首页的数据 (已完成)
export function getNav() {
  return request({
    url: "/api/sitenav/getnav",
    method: "get"
  });
}

// 创建接口（已完成）
export function createNavGroup(data) {
  return request({
    url: "/api/sitenav/group/add",
    method: "post",
    data
  });
}

// 更新接口（已完成）
export function updateNavGroup(data) {
  return request({
    url: "/api/sitenav/group/update",
    method: "post",
    data
  });
}

// 批量删除接口（已完成）
export function batchDeleteNavGroupByIds(data) {
  return request({
    url: "/api/sitenav/group/delete",
    method: "post",
    data
  });
}

// 创建接口（已完成）
export function createNavSite(data) {
  return request({
    url: "/api/sitenav/site/add",
    method: "post",
    data
  });
}

// 更新接口（已完成）
export function updateNavSite(data) {
  return request({
    url: "/api/sitenav/site/update",
    method: "post",
    data
  });
}

// 批量删除接口（已完成）
export function batchDeleteNavSiteByIds(data) {
  return request({
    url: "/api/sitenav/site/delete",
    method: "post",
    data
  });
}
